File: 1-100/94-binary-tree-inorder-traversal.cpp
// Recursive Solution
class Solution {
    vector<int> ans;
public:
    vector<int> inorderTraversal(TreeNode* root) {
        if (root == nullptr) return ans;
        helper(root);
        return ans;
    }

    void helper(TreeNode* root) {
        if (root == nullptr) return;
        helper(root->left);
        ans.push_back(root->val);
        helper(root->right);
    }
};

//Iteratively Version
class Solution {
    vector<int> ans;
public:
    vector<int> inorderTraversal(TreeNode* root) {
        if (root == nullptr) return ans;
        stack<TreeNode*> s;
        TreeNode* curr = root;
        s.push(root);
        curr = curr->left;
        while (curr || !s.empty()) {
            while (curr) {
                s.push(curr);
                curr = curr->left;
            }
            curr = s.top();
            s.pop();
            ans.push_back(curr->val);
            curr = curr->right;
        }

        return ans;
    }

};

File: python version/94-binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        self.ans = []

        def helper(root):
            if not root: 
                return
            helper(root.left)
            self.ans.append(root.val)
            helper(root.right)
        
        helper(root)
        
        return self.ans